x-pack/plugins/discover_log_explorer/public/hooks/use_control_panels.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ControlGroupInput } from '@kbn/controls-plugin/common';
+import { ControlGroupAPI } from '@kbn/controls-plugin/public';
+import { DataPublicPluginStart } from '@kbn/data-plugin/public';
+import { ViewMode } from '@kbn/embeddable-plugin/public';
+import { Query, TimeRange } from '@kbn/es-query';
+import { useQuerySubscriber } from '@kbn/unified-field-list';
+import { useSelector } from '@xstate/react';
+import { useCallback } from 'react';
+import { LogExplorerProfileStateService } from '../state_machines/log_explorer_profile';
+
+export const useControlPanels = (
+  logExplorerProfileStateService: LogExplorerProfileStateService,
+  data: DataPublicPluginStart
+) => {
+  const { query, filters, fromDate, toDate } = useQuerySubscriber({ data });
+  const timeRange: TimeRange = { from: fromDate!, to: toDate! };
+
+  const controlPanels = useSelector(logExplorerProfileStateService, (state) => {
+    if (!('controlPanels' in state.context)) return;
+    return state.context.controlPanels;
+  });
```

Comment:
```suggestion
  const controlPanels = useSelector(
    logExplorerProfileStateService,
    (state) => state.context.controlPanels
  );
```

Reply:
In case it's not present it will return `undefined`, so the same as having the empty return statement.